tests/src/test/scala/pathy/path.scala

```diff
@@ -100,133 +84,194 @@ class PathSpecs extends Specification with ScalaCheck {
+      }
 
-  "parseRelFile - foo/image.png" in {
-    parseRelFile("foo/image.png") must beSome(dir("foo") </> file("image.png"))
-  }
+      """preceeded with current directory such as "./image.png"""" in {
+        parseRelFile("./image.png") must beSome(file("image.png"))
+      }
 
-  "parseRelFile - ../foo/image.png" in {
-    parseRelFile("../foo/image.png") must beSome(currentDir <::> dir("foo") </> file("image.png"))
-  }
+      """path with two segments such as "foo/image.png"""" in {
+        parseRelFile("foo/image.png") must beSome(dir("foo") </> file("image.png"))
+      }
 
-  "parseRelFile - /foo/image.png" in {
-    parseRelFile("/foo/image.png") must beNone
-  }
+      """preceeded by a parent directory reference such as "../foo/image.png"""" in {
+        parseRelFile("../foo/image.png") must beSome(currentDir <::> dir("foo") </> file("image.png"))
+      }
+    }
 
-  "parseRelFile - foo/" in {
-    parseRelFile("foo/") must beNone
-  }
+    "should fail to parse" >> {
 
-  "parseAbsFile - /image.png" in {
-    parseAbsFile("/image.png") must beSome(rootDir </> file("image.png"))
-  }
+      """with a leading / such as "/foo/image.png"""" in {
+        parseRelFile("/foo/image.png") must beNone
+      }
 
-  "parseAbsFile - /foo/image.png" in {
-    parseAbsFile("/foo/image.png") must beSome(rootDir </> dir("foo") </> file("image.png"))
-  }
+      "with a trailing /" in {
+        parseRelFile("foo/") must beNone
+      }
 
-  "parseAbsFile - /foo/" in {
-    parseAbsFile("/foo/") must beNone
-  }
+      "." in {
+        parseRelFile(".") must beNone
+      }
 
-  "parseAbsFile - foo/image.png" in {
-    parseAbsFile("foo/image.png") must beNone
+      "foo/.." in {
+        parseRelFile("foo/..") must beNone
+      }
+    }
   }
 
-  "parseRelDir - empty string" in {
-    parseRelDir("") must beSome(currentDir[Unsandboxed])
-  }
+  "parseAbsFile" >> {
+    "should successfully parse" >> {
+      "a simple filename with extensions as long as there is a leading /" in {
+        parseAbsFile("/image.png") must beSome(rootDir </> file("image.png"))
+      }
 
-  "parseRelDir - ./../" in {
-    parseRelDir("./../") must beSome(currentDir <::> currentDir)
-  }
+      "a path with two segments" in {
+        parseAbsFile("/foo/image.png") must beSome(rootDir </> dir("foo") </> file("image.png"))
+      }
+    }
 
-  "parseRelDir - foo/" in {
-    parseRelDir("foo/") must beSome(dir("foo"))
-  }
+    "fail to parse" >> {
 
-  "parseRelDir - foo/bar/" in {
-    parseRelDir("foo/bar/") must beSome(dir("foo") </> dir("bar"))
-  }
+      "with a trailing /" in {
+        parseAbsFile("/foo/") must beNone
+      }
 
-  "parseRelDir - /foo/" in {
-    parseRelDir("/foo/") must beNone
-  }
+      "with no leading /" in {
+        parseAbsFile("foo/image.png") must beNone
+      }
 
-  "parseRelDir - foo" in {
-    parseRelDir("foo") must beNone
-  }
+      "/." in {
+        parseAbsFile("/.") must beNone
```

Comment:
This is one of the fixed cases, right? It would be easier to follow if the re-org was in a separate commit, I must say.